ggshield/core/url_utils.py

```diff
@@ -24,24 +24,33 @@ def clean_url(url: str, warn: bool = False) -> ParseResult:
     return parsed_url
 
 
-def dashboard_to_api_url(dashboard_url: str, warn: bool = False) -> str:
+def validate_instance_url(url: str, warn: bool = False) -> ParseResult:
     """
-    Convert a dashboard URL to an API URL.
-    handles the SaaS edge case where the host changes instead of the path
+    Validate a dashboard URL
     """
-    parsed_url = clean_url(dashboard_url, warn=warn)
+    parsed_url = clean_url(url, warn=warn)
     if parsed_url.scheme != "https" and not (
         parsed_url.netloc.startswith("localhost")
         or parsed_url.netloc.startswith("127.0.0.1")
     ):
-        raise UsageError(
-            f"Invalid scheme for dashboard URL '{dashboard_url}', expected HTTPS"
-        )
+        raise UsageError(f"Invalid scheme for dashboard URL '{url}', expected HTTPS")
     if any(parsed_url.netloc.endswith("." + domain) for domain in GITGUARDIAN_DOMAINS):
         if parsed_url.path:
```

Comment:
What would happen if a self hosted client had its instance be something like https://all-internal-tools.tech/gitguardian ?
My understanding it that he could be blocked because of the new URL validation ?

Reply:
Never mind it is only if we are on gitguardian instance because of the previous condition !